lib/src/Base/Func/SpecFunc/SpecFunc.cxx

```diff
@@ -745,6 +745,8 @@ Scalar SpecFunc::RegularizedIncompleteGammaInverse(const Scalar a,
 // Derivative of a Lanczos approximation of log(Gamma)
 Scalar SpecFunc::DiGamma(const Scalar x)
 {
+  // Throw if x is NaN
+  if (IsNaN(x)) throw InvalidArgumentException(HERE) << "Error: the argument of the DiGamma function is " << x;
```

Comment:
Use IsNormal to check inf too

Reply:
It already throws for invalid infinite values.

Reply:
where ? IsNaN does not checks for inf values

Reply:
I don't know, I simply ran the function on infinite values.

Reply:
NaN, on the other hand, does not throw with the current master so I patched this problem.

Reply:
As the code is valid only for normal numbers, you should use (!IsNormal(x)) even if it already throw for inf values somewhere else (in the x == round(x) test, which primary intention is to check if x is an integer, not an infinite number, see the message in the exception). It makes things much more readable.

Reply:
it's mostly best to catch invalid values early instead of wasting computations

Reply:
I disagree. *If it ain't broken, don't fix it!*